src/main/java/gregtech/api/pipenet/longdist/BlockLongDistancePipe.java

```diff
@@ -43,8 +45,25 @@ public void onBlockPlacedBy(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonn
         BlockPos.PooledMutableBlockPos offsetPos = BlockPos.PooledMutableBlockPos.retain();
         for (EnumFacing facing : EnumFacing.VALUES) {
             offsetPos.setPos(pos).move(facing);
-            LongDistanceNetwork network = LongDistanceNetwork.get(worldIn, offsetPos);
-            if (network != null && pipeType == network.getPipeType()) {
+            ILDNetworkPart networkPart;
+            IBlockState neighborState = worldIn.getBlockState(offsetPos);
+            if (neighborState.getBlock() instanceof ILDNetworkPart neighborBlock) {
+                networkPart = neighborBlock;
+            } else {
+                networkPart = ILDEndpoint.tryGet(worldIn, offsetPos);
```

Comment:
Since the LDEndpoint is an instanceof the ILDNetworkPart, why do you have the second check here? It seems like it would never get hit.

Reply:
Endpoints are MTEs so checking for block instanceof will never work for them